### PR TITLE
Small cleanup.

### PR DESCRIPTION
**Pull request by trivialfis**
* Add comment.
* Use threads from training parameter.
* Remove negative hessian handling.

### CODE REVIEW

**Comment by trivialfis** [comment]
Actually, I was able to produce 4.05 seconds runtime for gradient using setup from https://github.com/dmlc/xgboost/pull/7206#pullrequestreview-761240549 without using thread block.  I got the pointer outside the lambda function and pass them as lambda closure.  I will leave it as it's for now.  I plan to change the objective for multi-output regression and might go for my approach in the future. Hopefully, 0.5 seconds will be acceptable to you. ;-)

File: src/objective/regression_obj.cu [hunk 1]
```diff
@@ -107,13 +109,11 @@ class RegLossObj : public ObjFunction {
             }
             out_gpair_ptr[idx] = GradientPair(Loss::FirstOrderGradient(p, label) * w,
                                               Loss::SecondOrderGradient(p, label) * w);
-            if (out_gpair_ptr[idx].GetHess() < 0.0f) {
```
Comment (ShvetsKS) [inline src/objective/regression_obj.cu:110]: would it be better to add this check in next PRs as we can avoid extra gradient/hessian vector reading [here](https://github.com/dmlc/xgboost/blob/master/src/tree/updater_quantile_hist.cc#L458) ?

Comment (trivialfis) [reply src/objective/regression_obj.cu:110]: @ShvetsKS I think all the regression objectives defined by template have this check in their respective hessian function.  Could you please verify?

Comment (trivialfis) [reply src/objective/regression_obj.cu:110]: Specifically, they have something similar to `fmaxf(result, eps)` and eps is a positive value.

Comment (ShvetsKS) [reply src/objective/regression_obj.cu:110]: seems all `SecondOrderGradient`'s are bigger than zero: https://github.com/dmlc/xgboost/blob/4224c08cacceba3f83f90e387c07aa6205a83bfa/src/objective/regression_loss.h
as all results are returned with std::max(x, eps), where eps > 0;
and therefore check [here](https://github.com/dmlc/xgboost/blob/master/src/tree/updater_quantile_hist.cc#L483) is useless. So I'll just delete it in `updater_quantile_hist.cc` in next PR.

Thanks for providing the cleanup! I'll try to be more accurate.

Comment (trivialfis) [reply src/objective/regression_obj.cu:110]: I can't be sure about other objectives yet.  This is only for the objectives defined in that header.

Comment (trivialfis) [reply src/objective/regression_obj.cu:110]: Also, we should probably add a check to custom objectives provided by users before making that change.

**Pull request merged**